src/receipt-accumulator.ts

```diff
@@ -118,7 +118,7 @@ export class ReceiptAccumulator {
                             eventId,
                         };
 
-                        if (!data.thread_id || data.thread_id === MAIN_ROOM_TIMELINE) {
```

Comment:
A comment with some context would be helpful why "we shouldn't also be checking for `MAIN_ROOM_TIMELINE` because ... otherwise we run into bug x".

Reply:
That's not quite it - when the user marks a room as read e.g. by pressing Esc or clicking the "x" in the top right, we do send an unthreaded receipt. I'd happily merge a PR from you with a comment like this somewhere appropriate.

Reply:
Created https://github.com/matrix-org/matrix-js-sdk/pull/3378 and incorporated those caveats :+1: